### ISSUE TITLE
Make fail due to different declaration and error adding symbols: DSO missing from command line

### ISSUE THREAD

**Issue by sagitaninta** [grenaud/ROHan#10]
Hi Gabriel,

I was trying to install ROHan in another linux cluster in my own bin and the make command failed in this part:

```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
cd ../lib/samtools/ && make && cd ../..
make[2]: Entering directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/lib/samtools'
gcc -g -Wall -O2 -I. -I../htslib -I./lz4  -c -o bam2bcf.o bam2bcf.c
bam2bcf.c: In function ‘bcf_call_combine’:
bam2bcf.c:608:25: error: ‘FLT_MAX’ undeclared (first use in this function); did you mean ‘INT_MAX’?
             float min = FLT_MAX;
                         ^~~~~~~
                         INT_MAX
bam2bcf.c:608:25: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [Makefile:134: bam2bcf.o] Error 1
make[2]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/lib/samtools'
make[1]: *** [Makefile:46: ../lib/samtools/bedidx.o] Error 2
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

It runs again after I follow its suggestion to change the problematic FLT_MAX into INT_MAX but then the same problem appears here
```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
g++ -std=c++11  -Wall -lm -O3 -lz -I../lib/libgab/ -I../lib/libgab//gzstream/ -I../lib/samtools/ -I../lib/htslib/  -I../lib/libharu/include/   -c rohan.cpp -o rohan.o
rohan.cpp: In function ‘hResults computeLL(const std::vector<positionInformation>*, std::vector<PositionResult*>*, int)’:
rohan.cpp:1932:60: error: ‘LDBL_MAX’ was not declared in this scope
     long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
                                                            ^~~~~~~~
rohan.cpp:1932:60: note: suggested alternative: ‘LONG_MAX’
     long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
                                                            ^~~~~~~~
                                                            LONG_MAX
make[1]: *** [Makefile:13: rohan.o] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

And after again following the suggestions, make command exit with error status at this point. I don't now how to solve this.
```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
g++ -std=c++11  -o rohan rohan.o PdfWriter.o Hmm.o HmmState.o DataChunk.o DataToWrite.o GenoResults.o PositionResult.o miscfunc.o GenomicWindows.o GenomicRange.o ../lib/libgab/FastQObj.o ../lib/libgab/FastQParser.o ../lib/libharu/src/.libs/libhpdf.a ../lib/libgab/libgab.a ../lib/libgab/gzstream/libgzstream.a ../lib/samtools/bedidx.o ../lib/htslib/libhts.a ../lib/samtools/libbam.a ../lib/samtools/libst.a  -lgsl -lcurl -lgslcblas -llzma -lbz2 -lz  -lpthread 
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: ../lib/libharu/src/.libs/libhpdf.a(hpdf_fontdef_tt.o): undefined reference to symbol '__svml_idiv4'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/intel/19.0.5-gcc-uglchea/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64_lin/libsvml.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:28: rohan] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

I don't have admin privileges but I have checked that they have everything except for libpng-dev and autotools-dev but I have no idea how to get these and make rohan makefile read from there. Could you point me to a possible solution that not require me to contact the server's admin? Haha

Thanks for the cool software and have a good day,
Sabhrina

**Comment by sagitaninta** [comment]
Oh, it works after I change the bam2bcf.c at the error message location but then still error until here

```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
g++ -std=c++11  -Wall -lm -O3 -lz -I../lib/libgab/ -I../lib/libgab//gzstream/ -I../lib/samtools/ -I../lib/htslib/  -I../lib/libharu/include/   -c rohan.cpp -o rohan.o
rohan.cpp: In function ‘hResults computeLL(const std::vector<positionInformation>*, std::vector<PositionResult*>*, int)’:
rohan.cpp:1932:60: error: ‘LDBL_MAX’ was not declared in this scope
     long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
                                                            ^~~~~~~~
rohan.cpp:1932:60: note: suggested alternative: ‘LONG_MAX’
     long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
                                                            ^~~~~~~~
                                                            LONG_MAX
make[1]: *** [Makefile:13: rohan.o] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/software/rohan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

And I guess it will work if I change it again according to the suggestion.

**Comment by grenaud** [comment]
Hi Sabhrina, 
Sorry for the slugging response. Is it still not working? Because I just recompiled on my machine and it seems to be compiling/executing fine. On which OS is this? suse? 

Gabriel

**Comment by sagitaninta** [comment]
Hi Gabriel, no worries! I stupidly forgot to check the OS also, sorry. The Linux Cluster is under maintenance until tomorrow so I can only check this on Monday. I hope the maintenance somehow resolve this; I'll update you by then if it is working or not.

**Comment by sagitaninta** [comment]
Hi Gabriel, sorry for the delay as I was trying different ways to solve this.

After the cluster's update, I still cannot compile ROHan in it. I thus tried to transfer the compiled ROHan from the previous system I have been successfully running it with, and I think it manages to run with it. Please tell me this error means problem with the file and not with rohan 😂 

```
srun ~/bin/rohan/src/rohan -t 16 --rohmu 2e-5 -o CanidTrial_2e-5 --auto ${SCRATCH}/CanFam3.1.auto.txt ${SCRATCH}/Canis_lupus_familiaris.CanFam3.1.dna.toplevel.fa /dss/dssfs01/pn29bi/pn29bi-dss-0002/canids_bam/modern/C.mesomelas.realigned.bam
/dss/dsshome1/lxc06/ra57gih/bin/rohan/src/rohan: /usr/lib64/libcurl.so.4: no version information available (required by /dss/dsshome1/lxc06/ra57gih/bin/rohan/src/rohan)
Parsing arguments ......done
Computing average coverage..
ERROR: Cannot parse region "14:2000001-3000000"========================] 100 %
ERROR: Cannot parse region "16:7000001-8000000"
srun: error: i22r07c05s10: task 0: Segmentation fault (core dumped)
```

If the problem is with ROHan, I just tried to compile it again, but now in a conda environment, and it still have error like this
```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/ROHan/src'
cd ../lib/libgab/ &&  make libgab.a && make -C gzstream/ && cd ../..
make[2]: Entering directory '/dss/dsshome1/lxc06/ra57gih/ROHan/lib/libgab'
Makefile:72: warning: overriding recipe for target '-'
Makefile:68: warning: ignoring old recipe for target '-'
g++ -std=c++11  -Wall  -lm -O3 -I. -Igzstream/ -I/src/ -I/src/  -c libgab.cpp -o libgab.o
In file included from /dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/intel/19.0.5-gcc-uglchea/include/stdint.h:75,
                 from libgab.h:5,
                 from libgab.cpp:1:
/dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/gcc/8.4.0-gcc-656wch7/lib/gcc/x86_64-pc-linux-gnu/8.4.0/include/stdint.h:9:16: fatal error: stdint.h: No such file or directory
 # include_next <stdint.h>
                ^~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:100: libgab.o] Error 1
make[2]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/lib/libgab'
make[1]: *** [Makefile:53: ../lib/libgab/libgab.a] Error 2
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

Anyways, it actually runs, but from a previous compiling so the current compiling problem is still present. I really have no idea on how to solve this as this approach works with `bwa` (I can compile `bwa` after `git clone` in a separate conda environment).

As for the OS, I think this is what I have been using in the cluster:
```
cat /etc/os-release
NAME="SLES"
VERSION="15-SP1"
VERSION_ID="15.1"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP1"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp1"
```

I hope this is enough info for solving the compiling issue (?)

Best,
Sabhrina

**Comment by grenaud** [comment]
Hello! can you try doing a conda deactivate, recompile and launch again?

**Comment by sagitaninta** [comment]
I have recompile it when not in conda environment and that error previously is because I have not do `module load gsl `and all other necessary stuffs. Now it compiled but only until the same DSO error as previous first attempt. I have no idea at all what this error mean and how to solve it.

```
make -C  src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/ROHan/src'
g++ -std=c++11  -o rohan rohan.o PdfWriter.o Hmm.o HmmState.o DataChunk.o DataToWrite.o GenoResults.o PositionResult.o miscfunc.o GenomicWindows.o GenomicRange.o ../lib/libgab/FastQObj.o ../lib/libgab/FastQParser.o ../lib/libharu/src/.libs/libhpdf.a ../lib/libgab/libgab.a ../lib/libgab/gzstream/libgzstream.a ../lib/samtools/bedidx.o ../lib/htslib/libhts.a ../lib/samtools/libbam.a ../lib/samtools/libst.a  -lgsl -lcurl -lgslcblas -llzma -lbz2 -lz  -lpthread 
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: ../lib/libharu/src/.libs/libhpdf.a(hpdf_fontdef_tt.o): undefined reference to symbol '__svml_idiv4'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/intel/19.0.5-gcc-uglchea/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64_lin/libsvml.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:28: rohan] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

**Comment by grenaud** [comment]
Can you try:
g++ -std=c++11  -o rohan rohan.o PdfWriter.o Hmm.o HmmState.o DataChunk.o
DataToWrite.o GenoResults.o PositionResult.o miscfunc.o GenomicWindows.o
GenomicRange.o ../lib/libgab/FastQObj.o ../lib/libgab/FastQParser.o
../lib/libharu/src/.libs/libhpdf.a ../lib/libgab/libgab.a
../lib/libgab/gzstream/libgzstream.a ../lib/samtools/bedidx.o
../lib/htslib/libhts.a ../lib/samtools/libbam.a ../lib/samtools/libst.a
 -lgsl -lcurl -lgslcblas -llzma -lbz2 -lz  -lpthread  -lm

notice the -lm

**Comment by sagitaninta** [comment]
How do you actually do that?

I have ran this on the command line within `ROHan/src`:
`g++ -std=c++11  -o rohan rohan.o PdfWriter.o Hmm.o HmmState.o DataChunk.o DataToWrite.o GenoResults.o PositionResult.o miscfunc.o GenomicWindows.o GenomicRange.o ../lib/libgab/FastQObj.o ../lib/libgab/FastQParser.o ../lib/libharu/src/.libs/libhpdf.a ../lib/libgab/libgab.a ../lib/libgab/gzstream/libgzstream.a ../lib/samtools/bedidx.o ../lib/htslib/libhts.a ../lib/samtools/libbam.a ../lib/samtools/libst.a  -lgsl -lcurl -lgslcblas -llzma -lbz2 -lz  -lpthread -lm`

And got this 
```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: ../lib/libharu/src/.libs/libhpdf.a(hpdf_fontdef_tt.o): undefined reference to symbol '__svml_idiv4'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/intel/19.0.5-gcc-uglchea/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64_lin/libsvml.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by grenaud** [comment]
ok let me run some tests, I will get back to you!

**Comment by grenaud** [comment]
I just rebuilt it on a centos system. I do not know suse, is libtool
installed?

**Comment by sagitaninta** [comment]
I think so. I have to do `module load libtool` though.

```
libtool
Usage: /dss/dsshome1/lrz/sys/spack/staging/20.2.2/opt/x86_64/libtool/2.4.6-gcc-gdkhhgl/bin/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: you must specify a MODE
```

**Comment by sagitaninta** [comment]
The make just worked when I re-do everything in a new folder (now without conda), did you just change something in the current repo?

This is what I did for that:
```
module load git
module load autotools
module load gsl
module load gcc
module load cmake
module load libtool
git clone https://github.com/grenaud/ROHan.git
cd ROHan/
make
```

The first error message will be this:
```
make -C src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
cd ../lib/samtools/ && make && cd ../..
make[2]: Entering directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/lib/samtools'
gcc -g -Wall -O2 -I. -I../htslib -I./lz4 -c -o bam2bcf.o bam2bcf.c
bam2bcf.c: In function 'bcf_call_combine':
bam2bcf.c:608:25: error: 'FLT_MAX' undeclared (first use in this function); did you mean 'INT_MAX'?
float min = FLT_MAX;
^~~~~~~
INT_MAX
bam2bcf.c:608:25: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [Makefile:134: bam2bcf.o] Error 1
make[2]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/lib/samtools'
make[1]: *** [Makefile:46: ../lib/samtools/bedidx.o] Error 2
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

And then I do this to solve that:
```
vi lib/samtools/bam2bcf.c
# go to line 608 column 25 to change FLT_MAX to INT_MAX
# save the file
make
```

Then come the second error message:
```
gcc -L./lz4 -o samtools bam_index.o bam_plcmd.o sam_view.o bam_fastq.o bam_cat.o bam_md.o bam_reheader.o bam_sort.o bam_rmdup.o bam_rmdupse.o bam_mate.o bam_stat.o bam_color.o bamtk.o bam2bcf.o bam2bcf_indel.o sample.o cut_target.o phase.o bam2depth.o coverage.o padding.o bedcov.o bamshuf.o faidx.o dict.o stats.o stats_isize.o bam_flags.o bam_split.o bam_tview.o bam_tview_curses.o bam_tview_html.o bam_lpileup.o bam_quickcheck.o bam_addrprg.o bam_markdup.o tmp_file.o bam_ampliconclip.o amplicon_stats.o ./lz4/lz4.o libbam.a libst.a ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lcurses -lm -lz -lpthread
/usr/bin/ld: cannot find -lcurses
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:145: samtools] Error 1
make[2]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/lib/samtools'
make[1]: *** [Makefile:46: ../lib/samtools/bedidx.o] Error 2
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

I don't know how to solve that, so I run make again and the error message change:
```
make -C src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
g++ -std=c++11 -Wall -lm -O3 -lz -I../lib/libgab/ -I../lib/libgab//gzstream/ -I../lib/samtools/ -I../lib/htslib/ -I../lib/libharu/include/ -c rohan.cpp -o rohan.o
rohan.cpp: In function 'hResults computeLL(const std::vector<positionInformation>*, std::vector<PositionResult*>*, int)':
rohan.cpp:1932:60: error: 'LDBL_MAX' was not declared in this scope
long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
^~~~~~~~
rohan.cpp:1932:60: note: suggested alternative: 'LONG_MAX'
long double loglikelihoodForEveryPositionForEveryBaBdP=LDBL_MAX;
^~~~~~~~
LONG_MAX
make[1]: *** [Makefile:13: rohan.o] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

And solve that third error message like the first one:
```
vi src/rohan.cpp
# go to line 1932 column 60 to change LDBL_MAX to LONG_MAX
make
```

And there is the fourth error message:
```
make -C src/
make[1]: Entering directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
g++ -std=c++11 -Wall -lm -O3 -lz -I../lib/libgab/ -I../lib/libgab//gzstream/ -I../lib/samtools/ -I../lib/htslib/ -I../lib/libharu/include/ -c rohan.cpp -o rohan.o
rohan.cpp: In function 'hResults computeLL(const std::vector<positionInformation>*, std::vector<PositionResult*>*, int)':
rohan.cpp:2173:51: error: 'LDBL_MAX' was not declared in this scope
if(loglikelihoodForEveryPositionForEveryBaBdP == LDBL_MAX){//if previous hadn't been set
^~~~~~~~
rohan.cpp:2173:51: note: suggested alternative: 'LONG_MAX'
if(loglikelihoodForEveryPositionForEveryBaBdP == LDBL_MAX){//if previous hadn't been set
^~~~~~~~
LONG_MAX
make[1]: *** [Makefile:13: rohan.o] Error 1
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/test/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

And I did what I did with the third error message:
```
vi src/rohan.cpp
# go to line 2173 column 51
make
```

Aaaand it is done but with a lot of warning so I do not know if it is properly compiled considering that "`cannot find -lcurses`" I was not able to solve.

**Comment by grenaud** [comment]
regarding lcurses, can you have a look at the following:
https://stackoverflow.com/questions/33983504/gcc-ld-cant-find-lcurses
Tehre is an answer involving suse

**Comment by sagitaninta** [comment]
Hi Gabriel,

So apparently the problem lies in the cluster environment instead of ROHan. I managed to solve the issue with the IT and as I quote him:

> The issue arises from the fact that static libraries you are linking are built with intel modules and you are using gcc to make. I will try and solve the error and let you know. 
> 
> I recommend you first unload the intel module and then load gcc/8.4& gsl/2.5-gcc8. This should do the trick with the static library in lib/libharu/ compiled with gcc.

So I can compile the whole repo like this per his suggestion:
```
module load git
git clone https://github.com/grenaud/ROHan.git

module unload intel-mpi intel-mkl intel
module load gcc
module load  gsl/2.5-gcc8
cd ROHan
make
```

And then this stop at that -lcurses error, which is solved by

```
cd /lib/libharu
./configure
cd ../.. 
make
```
No need to change the bam2bcf.c file also as I did it previously.

So I am closing this and hope anyone can benefit from this solution somehow 😀

**Comment by grenaud** [comment]
thank you so much for raising this issue, this could indeed help others :-)

**Comment by sagitaninta** [comment]
I wanna notify here that this way of compiling ROHan does not work anymore in the same place after the cluster was upgraded. In fact, it won't work unless I specify gsl2.5 that has the library that ROHan need.

**Comment by grenaud** [comment]
Just to be sure, it does not work on gsl2.7?

**Comment by sagitaninta** [comment]
I have consulted with the IT so I think I have to revise: This is not because gsl2.7 but because the cluster don't have libcurses. This is what they said:

> I guess that libcurses is part of the termcap package, which is but not installed on our system. And I assume, we won't do so ... (as there are newer packages with the same or better functionality).

So when I installed it, I was using gsl2.5 so that version definitely need gsl2.5 and cannot work with libraries from gsl2.7. When I recompile it using the new gsl, it is stuck at here:

```
cd ../htslib && make lib-static
make[3]: Entering directory '/dss/dsshome1/lxc06/ra57gih/ROHan/lib/htslib'
make[3]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/lib/htslib'
gcc  -L./lz4  -o samtools bam.o bam_aux.o bam_index.o bam_plcmd.o sam_view.o bam_fastq.o bam_cat.o bam_md.o bam_plbuf.o bam_reheader.o bam_sort.o bam_rmdup.o bam_rmdupse.o bam_mate.o bam_stat.o bam_color.o bamtk.o bam2bcf.o sample.o cut_target.o phase.o bam2depth.o coverage.o padding.o bedcov.o bamshuf.o faidx.o dict.o stats.o stats_isize.o bam_flags.o bam_split.o bam_tview.o bam_tview_curses.o bam_tview_html.o bam_lpileup.o bam_quickcheck.o bam_addrprg.o bam_markdup.o tmp_file.o bam_ampliconclip.o amplicon_stats.o bam_import.o bam_samples.o bam_consensus.o consensus_pileup.o ./lz4/lz4.o libst.a ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lcurses -lm -lz  -lpthread
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lcurses
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:145: samtools] Error 1
make[2]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/lib/samtools'
make[1]: *** [Makefile:46: ../lib/samtools/bedidx.o] Error 2
make[1]: Leaving directory '/dss/dsshome1/lxc06/ra57gih/ROHan/src'
make: *** [Makefile:21: src/rohan] Error 2
```

The IT have provided me with a complicated workaround regarding that so this is more of cluster-specific problem rather than the software having the problem. Sorry for the false alarm.

**Comment by grenaud** [comment]
can we close the issue?
